File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/suggestions/SuggestionsFactoryTest.java
package coolpharaoh.tee.speicher.tea.timer.views.new_tea.suggestions;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class SuggestionsFactoryTest {

    @Test
    public void getBlackTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(0, null);
        assertThat(suggestions).isInstanceOf(BlackTeaSuggestions.class);
    }

    @Test
    public void getGreenTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(1, null);
        assertThat(suggestions).isInstanceOf(GreenTeaSuggestions.class);
    }

    @Test
    public void getYellowTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(2, null);
        assertThat(suggestions).isInstanceOf(YellowTeaSuggestions.class);
    }

    @Test
    public void getWhiteTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(3, null);
        assertThat(suggestions).isInstanceOf(WhiteTeaSuggestions.class);
    }

    @Test
    public void getOolongTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(4, null);
        assertThat(suggestions).isInstanceOf(OolongTeaSuggestions.class);
    }

    @Test
    public void getPuerhTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(5, null);
        assertThat(suggestions).isInstanceOf(PuerhTeaSuggestions.class);
    }

    @Test
    public void getHerbalTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(6, null);
        assertThat(suggestions).isInstanceOf(HerbalTeaSuggestions.class);
    }

    @Test
    public void getFruitTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(7, null);
        assertThat(suggestions).isInstanceOf(FruitTeaSuggestions.class);
    }

    @Test
    public void getRooibusTeaSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(8, null);
        assertThat(suggestions).isInstanceOf(RooibusTeaSuggestions.class);
    }

    @Test
    public void getNoSuggestions() {
        final Suggestions suggestions = SuggestionsFactory.getSuggestions(9, null);
        assertThat(suggestions).isInstanceOf(NoSuggestions.class);
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/suggestions/SuggestionsFactoryTest.kt
package coolpharaoh.tee.speicher.tea.timer.views.new_tea.suggestions

import android.app.Application
import coolpharaoh.tee.speicher.tea.timer.views.new_tea.suggestions.SuggestionsFactory.getSuggestions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class SuggestionsFactoryTest {
    @Test
    fun getBlackTeaSuggestions() {
        val suggestions = getSuggestions(0, Application())
        assertThat(suggestions).isInstanceOf(BlackTeaSuggestions::class.java)
    }

    @Test
    fun getGreenTeaSuggestions() {
        val suggestions = getSuggestions(1, Application())
        assertThat(suggestions).isInstanceOf(GreenTeaSuggestions::class.java)
    }

    @Test
    fun getYellowTeaSuggestions() {
        val suggestions = getSuggestions(2, Application())
        assertThat(suggestions).isInstanceOf(YellowTeaSuggestions::class.java)
    }

    @Test
    fun getWhiteTeaSuggestions() {
        val suggestions = getSuggestions(3, Application())
        assertThat(suggestions).isInstanceOf(WhiteTeaSuggestions::class.java)
    }

    @Test
    fun getOolongTeaSuggestions() {
        val suggestions = getSuggestions(4, Application())
        assertThat(suggestions).isInstanceOf(OolongTeaSuggestions::class.java)
    }

    @Test
    fun getPuerhTeaSuggestions() {
        val suggestions = getSuggestions(5, Application())
        assertThat(suggestions).isInstanceOf(PuerhTeaSuggestions::class.java)
    }

    @Test
    fun getHerbalTeaSuggestions() {
        val suggestions = getSuggestions(6, Application())
        assertThat(suggestions).isInstanceOf(HerbalTeaSuggestions::class.java)
    }

    @Test
    fun getFruitTeaSuggestions() {
        val suggestions = getSuggestions(7, Application())
        assertThat(suggestions).isInstanceOf(FruitTeaSuggestions::class.java)
    }

    @Test
    fun getRooibusTeaSuggestions() {
        val suggestions = getSuggestions(8, Application())
        assertThat(suggestions).isInstanceOf(RooibusTeaSuggestions::class.java)
    }

    @Test
    fun getNoSuggestions() {
        val suggestions = getSuggestions(9, Application())
        assertThat(suggestions).isInstanceOf(NoSuggestions::class.java)
    }
}